### ISSUE TITLE
[new input] touchscreen as hand tool doesn't work in Windows

### ISSUE THREAD

**Issue by cbm755** [xournalpp/xournalpp#1236]
**Affects versions :**
 - Windows
 - 1.0.12

**Describe the bug**
After enabling the new input system, I cannot make the touchscreen use the hand tool.  I think I could generally make this work in 1.0.10 (although it wasn't reliable and often started ignoring the touchscreen completely after some use, so I was hoping to experiment with the new input system).

Its not clear (to me) if the Preferences->Touchscreen tab is functional when I've enabled the New Input System.  Or am I supposed to configure inputs in some other way when using the New Input System.

**To Reproduce**
1. enable New Input System
2. Search for way to configure "touchscreen as hand tool" but cannot find.
3. "Touchscreen as hand tool" doesnt work OOTB (instead, it generally uses the selected tool).

**Input Logs**
"Follow this tutorial..."  Unfortunately, I don't know how to do any of this on Windows :(

**Comment by cbm755** [comment]
I got the machine working with Fedora 30.  With git master, the touchscreen works as handtool OOTB.

In preferences->input system, the "Input Devices" section is populated, whereas on Windows (or maybe just on 1.0.12) it says something about a placeholder.

**Comment by LittleHuba** [comment]
This is a duplicate of #1166. I'm currently working on a fix for that. I'll leave this ticket open as it has a more fitting title than #1166 and cares solely about that topic instead of mixing multiple ones.

**Comment by LittleHuba** [comment]
@cbm755 please test the linked PR #1242 if it resolves your issue.

Generally the touchscreen tab stays active even if the new input system is enabled. In the future some functionalities should be incorporated into the input system tab. For example assigning devices as pen when they are recognized as touchscreens is now handled in the input system tab.

**Comment by LittleHuba** [comment]
Please be aware that devices which were previously not listed have to be used once before opening the preferences to be listed.

**Comment by cbm755** [comment]
I'm willing to help but I have no dev skills (on Windows anyway).   I guess #1127 would help if CI runs produced .exe as an artifact... some day.

**Comment by LittleHuba** [comment]
Actually building on windows is not hard at all. Just follow these [instructions](https://github.com/xournalpp/xournalpp/blob/master/readme/WindowsBuild.md).

In the step "Get Sources and Build" 
you will have to use another repository to clone. 
https://github.com/LittleHuba/xournalpp.git 

And checkout  the branch `fix_device_listing` via `git checkout fix_device_listing`

Then you can follow the steps as listed in the document.

**Comment by Technius** [comment]
@cbm755 We now have automated nightly builds, so you can now download bleeding edge Windows builds from the releases page.

**Comment by cbm755** [comment]
@Technius that's great.  I will try that out later.

Is it possible to enable it for all PR, not just master branch?  Specifically, in this case @LittleHuba has asked me to test #1242...

**Comment by LittleHuba** [comment]
That is potentially possible but I am adverse to do this as the builds would nenn just to many jup handle. I'll make you a release the next few days by hand.

**Comment by cbm755** [comment]
nah, if its by hand anyway, I might as well do it.  I've got the azure yaml and the wiki

**Comment by cbm755** [comment]
I got your branch from #1242 compiled.  Running within the build tree I saw (hand typed b/c I can't figure out how to copy-paste from over the VNC connect I have to this machine):
```
** (xournalpp.exe:10604): WARNING **: ...  No device found. Is Xournal++ running in debugger / Eclipse...?
Probably this is the reason for not finding devices!

** (xournalpp.exe:10604): WARNING **: ...  No device found. Is Xournal++ running in debugger / Eclipse...?
Probably this is the reason for not finding devices!
```
From within Preferences -> Input System, under "Input Devices", it says:
```
No devices were found.  This seems wrong - maybe file <remaining cutoff>
```
Will try to get some debugging logs as per instructions elsewhere...

**Comment by cbm755** [comment]
[debug_input.log](https://github.com/xournalpp/xournalpp/files/3298409/debug_input.log)
[debug_gdk_input.log](https://github.com/xournalpp/xournalpp/files/3298411/debug_gdk_input.log)

These are now out of date: I will try again tonight with the updated commits in #1242.

**Comment by cbm755** [comment]
I tried with latest commits on #1242, same behaviour.

TL;DR: Windows sucks, as always.  Gtk3 for Windows is not well supported.  None of this seems to be Xournal++'s fault.

- - - -

  * I found I now have gtk3-demo installed: same problem there.  Inputs are:
      - `WACOM Tablet ISD Stylus`  (for pen)
      - `WACOM Tablet Eraser`  (for eraser button on back of pen).
      - `System Aggregrated Pointer` (for touch, so this is probably why it doesn't work properly in Xournal++.

  * played in "Device Manager" (some [Windows thing](https://support.microsoft.com/en-ca/help/4028019/windows-enable-and-disable-your-touchscreen-in-windows-10) and was able to update HID Wacom driver from 6/26/2015 to some later version (but still IIRC 2015).  Reboot.  This broke my stylus in Windows, so rolled back to 6/26/2015.

  * Looked at upstream GTK issues related to windows and input: not great, patches in limbo, lots of stuff related to "WinTab" API which is apparently not the most recent, etc.  The only people interested in "libinput"-style stuff on Windows are Xournal++, Inkscape and MyPaint developers.  This is all just AFAICT, but maybe useful as outsider's view.  Certainly not intended to offend anyone.

**Comment by LittleHuba** [comment]
Solved by #1318
